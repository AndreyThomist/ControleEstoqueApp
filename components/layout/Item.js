import React,{useCallback} from 'react'
import { View, Text, StyleSheet, Button, Image, TouchableOpacity } from 'react-native'
import Colors from '../../helpers/Colors'
import Card from '../ui/Card'
const Item = (props) => {
  const { navigation } = props;

  const detailScreenHandler = () => {
      navigation.navigate('DetailScreen', {
        name: props.name,
        imageUrl:props.imageUrl,
        quantity:props.quantity,
        id: props.id
      })
  }

  return (
    <Card style={styles.product} key={props.id} >
      <TouchableOpacity style={styles.touchable} onPress={detailScreenHandler}>
        <View style={styles.imageContainer}>
          <Image source={{ uri: props.imageUrl }} style={styles.image} />
        </View>
        <View style={styles.details}>
          <Text style={styles.name}>{props.name}</Text>
          <Text style={styles.quantity}><Text>Quantidade:</Text>{props.quantity}</Text>
        </View>
        <View style={styles.actions}>
          <Button title="Ver" onPress={detailScreenHandler} color={Colors.secondary} />
        </View>
      </TouchableOpacity>
    </Card>
  )
}

const styles = StyleSheet.create({
  product: {
    height: 250,
    margin: 20,
  },
  touchable: {
    borderRadius: 10,
    overflow: 'hidden',
    paddingBottom: 5
  },
  name: {
    fontSize: 17,
    fontFamily: 'exo2-bold',
    marginVertical: 2
  },
  imageContainer: {
    width: '100%',
    height: '60%',
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    overflow: 'hidden'
  },
  quantity: {
    fontFamily: 'exo2-italic'
  },
  image: {
    width: '100%',
    height: '100%'
  },
  details: {
    alignItems: 'center',
    height: '17%',
    padding: 10
  },
  actions: {
    marginTop: 5,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: '23%',
    paddingHorizontal: 20
  }
});

export default Item;