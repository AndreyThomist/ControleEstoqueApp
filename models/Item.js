class Item{
    constructor(id,imageUrl,userId,name,provider,quantity){
        this.id = id;
        this.imageUrl = imageUrl;
        this.userId = userId;
        this.name = name;
        this.provider = provider;
        this.quantity = quantity;
    }
}
export default Item;