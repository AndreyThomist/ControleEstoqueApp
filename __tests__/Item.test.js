import React from 'react'
import renderer from 'react-test-renderer'
import Item from '../components/layout/Item'

it('rederizando o componente item',() => {
    const tree = renderer.create(<Item  />).toJSON();
    expect(tree).toMatchSnapshot();
})