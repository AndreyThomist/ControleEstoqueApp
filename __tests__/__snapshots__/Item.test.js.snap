// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rederizando o componente item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "height": 250,
      "margin": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingBottom": 5,
      }
    }
  >
    <View
      style={
        Object {
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": "60%",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "17%",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "exo2-bold",
            "fontSize": 17,
            "marginVertical": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "exo2-italic",
          }
        }
      >
        <Text>
          Quantidade:
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "23%",
          "justifyContent": "center",
          "marginTop": 5,
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#5203fc",
                },
              ]
            }
          >
            Ver
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
